### ISSUE TITLE
About mini-batch in your model 

### ISSUE THREAD

**Issue by ChunchaoPeter** [cics-nd/gptorch#2]
Hi all,

I am currently reading your code. I am new to PyTorch and Computer. I am really like your work.

When I am reading the PyTorch online, I find that " torch.nn only supports mini-batches. The entire torch.nn package only supports inputs that are a mini-batch of samples, and not a single sample. "

When I reading your code, I find that your model inherited from the nn.Module.

class Model(torch.nn.Module):
    """
    Customized Model class for all GP objects
    """

Could you please tell me how can you address the mini-batches problem ?

Thanks in advance.

Best wishes,
Chunchao

**Comment by bmmi** [comment]
Hi Chunchao, thanks for reaching out. 
You are right, GP models usually compute gradients with all of the data (for small dataset), although some models do use stochastic gradient descent for big data, e.g. https://arxiv.org/abs/1309.6835. We primarily use PyTorch to store model parameters and compute gradients with `autograd`. And `autograd` works whether or not the input to the model is a single sample or a mini-batch of sample as long as the tensor dimension is legal to operate (also check [broadcast](https://pytorch.org/docs/stable/notes/broadcasting.html)). Just as the PyTorch tutorial says: "If you have a single sample, just use `input.unsqueeze(0)` to add a fake batch dimension." For GPs, the dimension of data is usually `[N, D]`, where `N` is the batch dim (num of data), `D` is the data dim.
Hope it helps.

**Comment by ChunchaoPeter** [comment]
Great! Thanks for your help.
It is really helpful!
